src/lints/trait_associated_const_added.ron

```diff
@@ -0,0 +1,62 @@
+SemverQuery(
+    id: "trait_associated_const_added",
+    human_readable_name: "non-sealed trait added associated constant without default value",
+    description: "A non-sealed trait has gained an associated constant without default value which break downstream implementations of the trait",
+    required_update: Major,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#trait-new-item-no-default"),
+    query: r#"
+    {
+        CrateDiff {
+            current {
+                item {
+                    ... on Trait {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        associated_constant {
+                            associated_constant: name @output @tag
+                            public_api_eligible @filter(op: "=", value: ["$true"])
+                            default @filter(op: "is_null") @output
+
+                            span_: span @optional {
+                               filename @output
+                               begin_line @output
+                            }
+                        }
+                    }
+                }
+            }
+            baseline {
+                item {
+                    ... on Trait {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        sealed @filter(op: "!=", value: ["$true"])
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        associated_constant @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+                            name @filter(op: "=", value: ["%associated_constant"])
+                            public_api_eligible @filter(op: "=", value: ["$true"])
+                            default @filter(op: "is_null")
```

Comment:
I think we don't want either of these two clauses.

We're looking for newly-added constants by a given name.

But currently, this block says: "there are no associated constants by that name that are public API (so not `#[doc(hidden)]`) and don't have a default value." So if the const exists but had a default value, this block will match. Same if the constant exists but is `#[doc(hidden)]`. In neither of those cases is the constant newly-added.

Could you add test cases for each of these cases and update the lint?